src/stores/AppStore.ts

```diff
@@ -2100,4 +2101,17 @@ export class AppStore {
     }
 
     // endregion
+
+    // Reset spectral profile's progress to 0 instead of cleaning the entire out-dated profile to avoid flashy effect in spectral profiler.
+    // Flashy effect: render empty profile and then render the coming profile, repeatedly.
+    public resetCursorRegionSpectralProfileProgress = (fileId: FileId) => {
+        this.spectralProfiles.get(fileId)?.get(CURSOR_REGION_ID)?.resetProfilesProgress();
+    };
+
+    public resetRegionSpectralProfileProgress = (regionId: RegionId) => {
+        const regionProfileStoreMaps = Array.from(this.spectralProfiles?.values());
```

Comment:
you can use a `for` loop here, looks neater:
`for (const [fileId, profile] of this.spectralProfiles)`

Otherwise, can't we also use `this.spectralProfiles?.forEach`?